jit/src/lib.rs

```diff
@@ -0,0 +1,146 @@
+use std::error::Error;
+use std::fmt;
+use std::mem;
+
+use cranelift::prelude::*;
+use cranelift_module::{Backend, FuncId, Linkage, Module, ModuleError};
+use cranelift_simplejit::{SimpleJITBackend, SimpleJITBuilder};
+
+use rustpython_bytecode::bytecode;
+
+mod instructions;
+
+use self::instructions::FunctionCompiler;
+
+#[derive(Debug)]
+pub enum JITCompileError {
+    NotSupported,
+    BadBytecode,
+    CraneliftError(ModuleError),
+}
+
+impl From<ModuleError> for JITCompileError {
+    fn from(err: ModuleError) -> Self {
+        JITCompileError::CraneliftError(err)
+    }
+}
+
+impl fmt::Display for JITCompileError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match *self {
+            JITCompileError::NotSupported => f.write_str("Function can't be jitted."),
+            JITCompileError::BadBytecode => f.write_str("Bad bytecode."),
+            JITCompileError::CraneliftError(ref err) => err.fmt(f),
+        }
+    }
+}
+
+impl Error for JITCompileError {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        match *self {
+            JITCompileError::CraneliftError(ref err) => Some(err),
+            _ => None,
+        }
+    }
+}
+
+struct JIT {
```

Comment:
Rust type names are camel case, including for acronyms -- if you want, you could `s/JIT/Jit` for the type names